objectherkenning_openbare_ruimte/databricks_pipelines/post_processing_pipeline/data_enrichment_step/components/clustering_detections.py

```diff
@@ -0,0 +1,200 @@
+import numpy as np
+from databricks.sdk.runtime import sqlContext
+from pyspark.sql import SparkSession, Window
+from pyspark.sql.functions import col, mean, monotonically_increasing_id, row_number
+from shapely.geometry import Point
+from sklearn.cluster import DBSCAN
+
+MS_PER_RAD = 6371008.8  # Earth radius in meters
+MIN_SAMPLES = 1  # avoid noise points. All points are either in a cluster or are a cluster of their own
+
+
+class Clustering:
+
+    def __init__(self, spark: SparkSession, catalog, schema):
+
+        self.spark = spark
+        self.catalog = catalog
+        self.schema = schema
+        query_detection_metadata = f"SELECT * FROM {self.catalog}.{self.schema}.silver_detection_metadata WHERE status='Pending'"
```

Comment:
Would it make sense maybe to move all queries to a class? Maybe the table manager or maybe a subclass?
Or maybe have different data model classes where each one has the methods for that table and details about it.
If you want we can also do a refactoring later and discuss how to do it 😉

Reply:
i was wondering the same, but there are so many different queries

for me it made sense to add the common ones to TableManager and keep the rest aside